docs/website/docs/walkthroughs/deploy-a-pipeline/deploy-with-google-cloud-functions.md

```diff
@@ -0,0 +1,77 @@
+    ```python
+    from notion_pipeline import load_databases
+
+    def pipeline_notion(request):
+      load_databases()
+      return "Pipeline run successfull!"
+    ```
+    By default, Google Cloud Functions looks for the main.py file in the main directory, and we called the `load_databases()` function from notion_pipeline.py as shown above.
+
+## 2. Deploying GCP Cloud Function
+In a shell editor, navigate to the directory where the "main.py" file is located and run the following command in the terminal
+```bash
+ gcloud functions deploy pipeline_notion --runtime python310 --trigger-http --allow-unauthenticated --source . --timeout 300
+
+```
+        
+- This command deploys a function called "pipeline_notion" with Python 3.10 as the runtime environment, an HTTP trigger, and allows unauthenticated access. The source "." refers to all files in the directory. The timeout is set to 5 mins ( 300 secs ), 
+- If you are loading a large number of files to the destination you can increase this to 60 minutes for HTTP functions. 10 minutes for event driven functions. To learn more about the function timeout, you can read the [documentation here.](https://cloud.google.com/functions/docs/configuring/timeout)
+
+  
+## 3. Setting up environmental variables in the Cloud Function
+Environmental variables can be declared in the Cloud Function in two ways:
+
+#### 3a. Directly in the function:
+
+- Go to the Google Cloud Function and select the deployed function. Click 'EDIT'.
+- Navigate to the 'BUILD' tab and click 'ADD VARIABLE' under 'BUILD ENVIRONMENTAL VARIABLE'.
+- Enter a name for the variable that corresponds to the argument required by the pipeline. Make sure to capitalize the variable name if it is specified in secrets.toml. For example, if the variable name is api_key, set the variable name to "API_KEY".
+- Enter the value for the Notion API key.
+- Click Next and deploy the function.
+
+#### 3b. Use GCP Secret Manager:
+
+- Go to the Google Cloud function and select the function you deployed. Click 'EDIT'.
+- In the 'Runtime, Build, Connections and Security Settings' section, select 'Security and Images Repo'.
+- Click 'Add a secret reference' and select the secret you created, for example 'notion_secret'.
+- Set the 'Reference method' to 'Mounted as environment variable'.
+- In the 'Environment Variable' field, enter the name of the environment variable that corresponds to the argument required by the pipeline. - Remember to capitalize the variable name if it is required by the pipeline and specified in secrets.toml. For example, if the variable name is api_key and you have defined the value in the secrets manager secret as "notion_secret", you would declare the environment variable as "API_KEY".
+- Finally, click 'DEPLOY' to deploy the function. The HTTP trigger will now successfully execute the pipeline every time the URL is triggered.
```

Comment:
here the info on adding permissions to read (Secret Accessor) is actually useful :) sorry for being pedantic. but I'd like to have a good working deployment because we'll implement it in `dlt` core soon

Reply:
Added Sir, please see.